File: backend/internal/model/problem.go
package model

type Problem struct {
	ID          string   `json:"id"`
	Title       string   `json:"title"`
	Contest     string   `json:"contest"`
	Difficulty  int      `json:"difficulty"`
	Tags        []string `json:"tags"`
	AtCoderURL  string   `json:"atCoderUrl"`
	ProblemsURL string   `json:"problemsUrl"`
	Solved      bool     `json:"solved"`
	Favorite    bool     `json:"favorite"`
}

File: frontend/src/types/problem.ts
export interface Tag {
  id: number
  name: string
}

export interface Problem {
  id: string
  title: string
  contest: string
  difficulty: number
  tags: Tag[]
  atCoderUrl: string
  problemsUrl: string
  solved?: boolean
  favorite?: boolean
}

export interface UserProblemStatus {
  problemId: string
  userId: string
  solved: boolean
  favorite: boolean
  memo: string
  updatedAt: number
}